src/harness/testcases/WINNF_FT_S_PCR_testcase.py

```diff
@@ -0,0 +1,263 @@
+    device_a['installationParam']['eirpCapability'] = 30
+    device_b['installationParam']['eirpCapability'] = 47
+    device_a['installationParam']['antennaGain'] = 16
+    device_b['installationParam']['antennaGain'] = 16
+
+    conditionals_b = {
+        'cbsdCategory': device_b['cbsdCategory'],
+        'fccId': device_b['fccId'],
+        'cbsdSerialNumber': device_b['cbsdSerialNumber'],
+        'airInterface': device_b['airInterface'],
+        'installationParam': device_b['installationParam'],
+        'measCapability': device_b['measCapability']
+    }
+    conditionals = {'registrationData': [conditionals_b]}
+    del device_b['installationParam']
+    del device_b['cbsdCategory']
+    del device_b['airInterface']
+    del device_b['measCapability']
+
+    # Create the actual config.
+    devices = [device_a, device_b]
+    config = {
+        'registrationRequests': devices,
+        'conditionalRegistrationData': conditionals,
+        'palRecords': pal_records,
+        'ituDataPath': CONFIG.GetItuDir(),
+        'terrainDataPath': CONFIG.GetTerrainDir(),
+        'landCoverDataPath': CONFIG.GetLandCoverDir(),
+        'serverCert': os.path.join('certs/sas.cert'),
+        'serverKey': os.path.join('certs/sas.key')
+    }
+    writeConfig(filename, config)
+
+  @configurable_testcase(generate_PCR_1_default_config)
+  def test_WINNF_FT_S_PCR_1(self, config_filename):
+    """ Successful Maximum PPA Creation
+
+    Checks PPA generated by SAS UUT shall be fully contained within the service area.
+    """
+    # Load the Config file
+    config = loadConfig(config_filename)
+    pal_records = config['palRecords']
+    itu_data_path = config['ituDataPath']
+    terrain_data_path = config['terrainDataPath']
+    landcover_data_path = config['landCoverDataPath']
+
+    # light checking of itu,terrain and landcover data path exists
+    self.assertTrue(os.path.exists(itu_data_path), msg='ITU Data path is not configured')
+    self.assertTrue(os.path.exists(terrain_data_path), msg='Terrain Data path is not configured')
+    self.assertTrue(os.path.exists(landcover_data_path), msg='LandCover Data path is not configured')
+
+    # Whitelist user IDs.
+    for device in config['registrationRequests']:
+      self._sas_admin.InjectFccId({'fccId': device['fccId']})
+      self._sas_admin.InjectUserId({'userId': device['userId']})
+
+    # Inject the PAL records
+    for pal_record in pal_records:
+      self._sas_admin.InjectPalDatabaseRecord(pal_record)
+
+    # Pre-load conditional registration data for N3 CBSDs.
+    if ('conditionalRegistrationData' in config) and \
+            (config['conditionalRegistrationData']):
+      self._sas_admin.PreloadRegistrationData(
+          config['conditionalRegistrationData'])
+
+    # Register devices and  check response.
+    cbsd_ids = self.assertRegistered(config['registrationRequests'])
+
+    # Trigger SAS UUT to create a PPA boundary
+    pal_ids = []
```

Comment:
You can do simply

```
pal_ids = [record['palId'] for record in pal_records]
```

Reply:
Thanks.   we have made the changes as suggested and tested it.